pkg/endpoint/policy.go

```diff
@@ -709,3 +710,48 @@ func (e *Endpoint) GetCIDRPrefixLengths() (s6, s4 []int) {
 	}
 	return e.desiredPolicy.CIDRPolicy.ToBPFData()
 }
+
+// UpdateVisibilityPolicy updates the visibility policy of this endpoint to
+// reflect the state stored in the proxy visibility annotation, if present,
+// in anno. If no such annotation is present, then the VisibilityPolicy for the
+// Endpoint will be empty, and will have no effect. If the proxy visibility
+// annotation cannot be parsed, an empty visibility policy is assigned to the
+// Endpoint.
+func (e *Endpoint) UpdateVisibilityPolicy(anno map[string]string) {
```

Comment:
nit: This is half-way between being explicit about policy and being generic to handle any annotations. Do you expect that the endpoint package should be responsible for understanding specific annotations going forward?

A couple of suggestions:
`UpdateAnnotations(anno map[string]string)` would be more general and establish that `pkg/endpoint` should know about annotation keys; alternatively,
`UpdateVisibilityPolicy(annotation string)` would be more specific (callers would call `UpdateVisibilityPolicy(annotations[annotation.ProxyVisibility])`), which would allow the endpoint package to be oblivious of the annotation key value and not depend on `pkg/annotation`.

Reply:
I'll implement it as you like. I personally err on the side of having more generic functionality in the parameter list for functions to avoid having to refactor all callsites in the future; if we add another annotation we care about, we get it for free in the current form. But, I don't really have a preference so I will defer to you here.

Reply:
If you want it to be generic to reduce need for refactoring in future, then `UpdateAnnotations()` would make more sense. If we left it as-is, then in future you'd need to at least rename the function anyway because most likely any new annotations would have their own semantics and potentially wouldn't even be related to visibility policy at all.